pkg/endpoint/endpoint.go

```diff
@@ -2295,3 +2304,40 @@ func (e *Endpoint) PinDatapathMap() error {
 
 	return err
 }
+
+func (e *Endpoint) syncEndpointHeaderFile() {
+	e.BuildMutex.Lock()
+	defer e.BuildMutex.Unlock()
+
+	if err := e.LockAlive(); err != nil {
+		// endpoint was removed in the meanwhile, return
+		return
+	}
+	defer e.Unlock()
+
+	e.writeHeaderfile(e.StateDirectoryPath())
+}
+
+// SyncEndpointHeaderFile it bumps the current DNS History information for the
+// endpoint in the lxc_config.h file.
+func (e *Endpoint) SyncEndpointHeaderFile() error {
+	e.UnconditionalLock()
```

Comment:
Can we unconditionally lock here? Sure we looked up the endpoint but could it also disappear?